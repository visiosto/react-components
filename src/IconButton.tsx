// Copyright (c) 2021 Visiosto oy
// Licensed under the MIT License

import React from 'react';

import Button, { ButtonProps } from './Button';
import Icon, { Icons } from './Icon';

export type IconButtonProps = ButtonProps & { icon: Icons };

function IconButton({
  action = 'default',
  children,
  icon,
  onClick = undefined,
  thickness = 'default',
}: IconButtonProps) {
  return (
    <Button action={action} onClick={onClick} thickness={thickness}>
      <Icon icon={icon} /> {children}
    </Button>
  );
}

export default IconButton;
