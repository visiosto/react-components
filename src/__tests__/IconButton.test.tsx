// Copyright (c) 2021 Visiosto oy
// Licensed under the MIT License

import React from 'react';

import IconButton from '../IconButton';

import renderWithProviders from '../util/test/renderWithProviders';

describe('Icon button component', () => {
  it('renders default colours correctly', () => {
    const { container, getByText } = renderWithProviders(
      <IconButton icon="cancel">IconButton</IconButton>,
    );

    expect(container).toMatchSnapshot();

    expect(getByText('IconButton')).toBeInTheDocument();
  });

  it('renders accept colours correctly', () => {
    const { container, getByText } = renderWithProviders(
      <IconButton action="accept" icon="search">
        IconButton
      </IconButton>,
    );

    expect(container).toMatchSnapshot();

    expect(getByText('IconButton')).toBeInTheDocument();
  });

  it('renders cancel colours correctly', () => {
    const { container, getByText } = renderWithProviders(
      <IconButton action="cancel" icon="navigate_next">
        IconButton
      </IconButton>,
    );

    expect(container).toMatchSnapshot();

    expect(getByText('IconButton')).toBeInTheDocument();
  });

  it('renders default colours correctly on a thin button', () => {
    const { container, getByText } = renderWithProviders(
      <IconButton icon="tune" thickness="thin">
        IconButton
      </IconButton>,
    );

    expect(container).toMatchSnapshot();

    expect(getByText('IconButton')).toBeInTheDocument();
  });

  it('renders accept colours correctly on a thin button', () => {
    const { container, getByText } = renderWithProviders(
      <IconButton action="accept" icon="language" thickness="thin">
        IconButton
      </IconButton>,
    );

    expect(container).toMatchSnapshot();

    expect(getByText('IconButton')).toBeInTheDocument();
  });

  it('renders cancel colours correctly on a thin button', () => {
    const { container, getByText } = renderWithProviders(
      <IconButton action="cancel" icon="event" thickness="thin">
        IconButton
      </IconButton>,
    );

    expect(container).toMatchSnapshot();

    expect(getByText('IconButton')).toBeInTheDocument();
  });
});
