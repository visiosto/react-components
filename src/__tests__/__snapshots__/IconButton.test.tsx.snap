// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon button component renders accept colours correctly 1`] = `
<div>
  <span
    class="Button__Span-sc-o80e68-0 Button__SpanAccept-sc-o80e68-3 hbAGrJ jelbpC"
  >
    <span
      class="Icon__Span-sc-1qcplrz-0 kqLtxw material-icons md-24 IconButton__ButtonIcon-sc-1o6fqy9-0 jqJFXe"
      size="24"
    >
      search
    </span>
     
    IconButton
  </span>
</div>
`;

exports[`Icon button component renders accept colours correctly on a thin button 1`] = `
<div>
  <span
    class="Button__Span-sc-o80e68-0 Button__SpanAccept-sc-o80e68-3 hbAGrJ jelbpC"
  >
    <span
      class="Icon__Span-sc-1qcplrz-0 kqLtxw material-icons md-24 IconButton__ButtonIcon-sc-1o6fqy9-0 jqJFXe"
      size="24"
    >
      language
    </span>
     
    IconButton
  </span>
</div>
`;

exports[`Icon button component renders cancel colours correctly 1`] = `
<div>
  <span
    class="Button__Span-sc-o80e68-0 Button__SpanCancel-sc-o80e68-4 hbAGrJ bTSMet"
  >
    <span
      class="Icon__Span-sc-1qcplrz-0 kqLtxw material-icons md-24 IconButton__ButtonIcon-sc-1o6fqy9-0 jqJFXe"
      size="24"
    >
      navigate_next
    </span>
     
    IconButton
  </span>
</div>
`;

exports[`Icon button component renders cancel colours correctly on a thin button 1`] = `
<div>
  <span
    class="Button__Span-sc-o80e68-0 Button__SpanCancel-sc-o80e68-4 hbAGrJ bTSMet"
  >
    <span
      class="Icon__Span-sc-1qcplrz-0 kqLtxw material-icons md-24 IconButton__ButtonIcon-sc-1o6fqy9-0 jqJFXe"
      size="24"
    >
      event
    </span>
     
    IconButton
  </span>
</div>
`;

exports[`Icon button component renders default colours correctly 1`] = `
<div>
  <span
    class="Button__Span-sc-o80e68-0 Button__SpanDefault-sc-o80e68-2 hbAGrJ hWgwyh"
  >
    <span
      class="Icon__Span-sc-1qcplrz-0 kqLtxw material-icons md-24 IconButton__ButtonIcon-sc-1o6fqy9-0 jqJFXe"
      size="24"
    >
      cancel
    </span>
     
    IconButton
  </span>
</div>
`;

exports[`Icon button component renders default colours correctly on a thin button 1`] = `
<div>
  <span
    class="Button__SpanThin-sc-o80e68-1 Button__SpanDefaultThin-sc-o80e68-5 jlfLWh eQMiB"
  >
    <span
      class="Icon__Span-sc-1qcplrz-0 kqLtxw material-icons md-24 IconButton__ButtonIcon-sc-1o6fqy9-0 jqJFXe"
      size="24"
    >
      tune
    </span>
     
    IconButton
  </span>
</div>
`;
